lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,792 @@
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <cmath>
+#include "openturns/PointConditionalDistribution.hxx"
+#include "openturns/IteratedQuadrature.hxx"
+#include "openturns/ParametricFunction.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/ResourceMap.hxx"
+#include "openturns/SimplicialCubature.hxx"
+#include "openturns/CubaIntegration.hxx"
+#include "openturns/Tuples.hxx"
+#include "openturns/Normal.hxx"
+#include "openturns/Student.hxx"
+#include "openturns/OptimizationAlgorithm.hxx"
+#include "openturns/RandomGenerator.hxx"
+#include "openturns/DomainEvent.hxx"
+#include "openturns/PlatformInfo.hxx"
+#include "openturns/JointDistribution.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(PointConditionalDistribution)
+
+static const Factory<PointConditionalDistribution> Factory_PointConditionalDistribution;
+
+/* Default constructor */
+PointConditionalDistribution::PointConditionalDistribution()
+  : DistributionImplementation()
+{
+  setName("PointConditionalDistribution");
+  setDimension(1);
+  update();
+}
+
+/* Parameters constructor */
+PointConditionalDistribution::PointConditionalDistribution(const Distribution & distribution,
+    const Indices & conditioningIndices,
+    const Point & conditioningValues)
+  : DistributionImplementation()
+  , distribution_(distribution)
+  , conditioningIndices_(conditioningIndices)
+  , conditioningValues_(conditioningValues)
+{
+  setName("PointConditionalDistribution");
+  const UnsignedInteger fullDimension = distribution.getDimension();
+  if (!distribution.isDiscrete() && !distribution.isContinuous())
+    throw NotYetImplementedException(HERE) << "PointConditionalDistribution for mixed continuous/discrete case";
+  if (conditioningIndices.getSize() != conditioningValues.getSize())
+    throw InvalidArgumentException(HERE) << "Conditioning indices/values sizes do not match, got " << conditioningIndices.getSize() << "/" << conditioningValues.getSize();
+  if (!conditioningIndices.check(fullDimension))
+    throw InvalidArgumentException(HERE) << "Conditioning vector indices (" << conditioningIndices << ") must be less than conditioned distribution dimension (" << fullDimension <<" )";
+
+  if (conditioningIndices.getSize())
+  {
+    const Interval conditioningRange(distribution.getMarginal(conditioningIndices).getRange());
+    if (!conditioningRange.contains(conditioningValues))
+      throw InvalidArgumentException(HERE) << "Conditioning vector (" << conditioningValues << ") not in conditioning range (" << conditioningRange << ")";
+  }
+
```

Comment:
As cuba defaults to vegas in the 1D case, which is a sampling algorithm and as such a slowly converging method, it would be better to use GaussKronrod in the case dimension=1:
```
if (fullDimension == conditioningIndices.getSize() + 1)
{
  integrationAlgorithm_ = GaussKronrod();
}
else
{
  // The current code from line 79 to 89
}
update();
```

Reply:
good idea